### DISCUSSION TITLE
How to see failed ICV verification in StrongSwan log ?

### DISCUSSION THREAD

**Discussion by jcdemay** [strongswan/strongswan#1933] [General]
When using the userland libipsec module, I can see the corrupted packets that have been received:
Sep 28 08:47:58 04[ESP] ESP decryption or ICV verification failed.

How can I get the same information in the log file *without* using the libipsec module ? I really wish to be able to monitor such event but I also wish to not rely on the userland module for that, using the Linux kernel internal IP framework is kind of mandatory for good performance and efficiency.

**Comment by tobiasbrunner** [comment]
> How can I get the same information in the log file _without_ using the libipsec module ?

What do you mean? From the Linux kernel? There is no such log message there (unless you add one). The only thing you'll see are increased error counters (in `/proc/net/xfrm_stat` and `ip -s xfrm state`, depending on the error).

**Reply by jcdemay** [reply]
Yes from the kernel. I know of /proc/net/xfrm_stat but that is just a counter and the thing is, when an ICV verification has failed, there might be something really wrong to investigate, thus the fact that I would like to know at least when and from which IP address the corrupted packet originates from.

Maybe through the Linux audit subsystem ?

**Reply by tobiasbrunner** [reply]
What do you want to investigate exactly?

> Maybe through the Linux audit subsystem ?

There actually is an audit message if the ICV verification fails (SA-icv-failure).

**Reply by jcdemay** [reply]
_**"There actually is an audit message if the ICV verification fails (SA-icv-failure)."**_
Yep, just found it in net/xfrm/xfrm_state.c
According to uapi/linux/audit.h, the message type to log is MAC_IPSEC_EVENT.
